step-functions/step-functions-handler/src/main/java/step/functions/handler/FunctionMessageHandler.java

```diff
@@ -181,45 +186,19 @@ private LiveReporting initializeLiveReporting(Map<String, String> properties) th
 			// We currently only support Websocket uploads; if this changes in the future, here is the place to modify the logic.
 			String uploadContextId = properties.get(StreamingResourceUploadContext.PARAMETER_NAME);
 			if (uploadContextId != null) {
-				String host; // actually contains scheme, hostname, and potentially port.
-				// If present, agent-side configuration overrides the default value.
-				// Note: in case we use the URL in other places, this should probably be put somewhere else (maybe
-				// LiveReporting itself), but that's for the future.
-				// Note: both agentProperties or the value might be undefined.
-				String agentConfUrl = Optional.ofNullable(agentTokenServices.getAgentProperties())
-						.map(m -> m.get("step.reporting.url"))
-						.orElse(null);
-				if (agentConfUrl != null) {
-					// just a sanity check really
-					if (!agentConfUrl.matches("^https?://.+")) {
-						throw new IllegalArgumentException("Invalid URL in 'step.reporting.url' (agent-side configuration): " + agentConfUrl);
+				URI websocketUploadUri = getWebsocketUploadUri(properties, uploadContextId);
+				@SuppressWarnings("unchecked") Class<StreamingUploadProvider> providerClass = (Class<StreamingUploadProvider>) Thread.currentThread().getContextClassLoader().loadClass("step.streaming.websocket.client.upload.WebsocketUploadProvider");
+				if (webSocketContainer == null) {
+					synchronized(FunctionMessageHandler.this) {
+						if (webSocketContainer == null) {
+							webSocketContainer = providerClass.getDeclaredMethod("instantiateWebSocketContainer").invoke(null);

```

Comment:
If instantiateWebSocketContainer would be part of the interface StreamingUploadProvider there would be no need for reflection

Reply:
I actually tried that, but then I got ClassCastExceptions all over the place because the classes loaded from the two classloaders were deemed incompatible, so I just left this as is.